version: "3.7"

x-logging: &logging
  logging:
    driver: "json-file"
    options:
      max-file: "5"
      max-size: "10m"

x-restart-policy:
  restart: unless-stopped

services:
  adminer:
    <<: *logging

    image: "docker.io/adminer:latest"

    ports:
    - "8080:8080"

  postgres:
    <<: *logging

    image: "docker.io/postgres:16-alpine"

    ports:
    - "5432:5432"

    environment:
      POSTGRES_DB: xythrion
      POSTGRES_PASSWORD: xythrion
      POSTGRES_USER: xythrion

    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U xythrion"]
      interval: 2s
      timeout: 1s
      retries: 5

  api:
    <<: *logging

    build:
      context: ./api
      dockerfile: Dockerfile

    env_file:
    - .env

    environment:
      XYTHRION_API_RELOAD: "false"
      XYTHRION_API_DB_HOST: "postgres"
      XYTHRION_API_HOST: "0.0.0.0"

    volumes:
    - ./api:/api

    depends_on:
    - postgres

    ports:
    - "8001:8001"


  bot:
    <<: *logging

    build:
      context: ./bot
      dockerfile: Dockerfile

    environment:
      API_URL: "http://api:8001"

    env_file:
    - .env

    depends_on:
    - api
    - postgres

    volumes:
    - ./bot:/bot
